packages/bsky/src/hydration/hydrator.ts

```diff
@@ -514,6 +514,7 @@ export class Hydrator {
       const parent = replies.get(uri)
       if (!parent?.record.reply) return
       replyParentAuthors.push(didFromUri(parent.record.reply.parent.uri))
+      postAndReplyRefs.push(parent.record.reply.parent)
```

Comment:
Is this still needed?

Reply:
I see now that this is probably what guarantees that post block data between a post's parent and root gets hydrated.  It's a pretty sneaky way to get that though, and it's going to kick off a fair amount of other work along the way.  I think getting more explicit with it in `hydratePostBlocks` is probably the way to go.

Reply:
Nope was redundant

Reply:
I think the work is already being done here 🤔 https://github.com/bluesky-social/atproto/blob/reply-grandparent/packages/bsky/src/hydration/hydrator.ts#L504

Reply:
I think that helps us cover the case `A -> B -> C`, but in the case `A -> ... -> C -> D`, there's nothing that checks the 3p blocks between `C` (`D`'s parent) and `A` (`D`'s root).

Reply:
In the case `A -> B -> C` it works because you hydrate `B` and therefore check its 3p blocks with its parent `A`, which just so happens to be `C`'s root.

Reply:
I think actually that in the case of `A -> ... -> C -> D`, we do hydrate post blocks for `A <-> C`, since `C` is hydrated due to its presence on `D.reply.parent`. Added a test for this, where `D.reply.root` is a blocked post because `A` blocks `C` https://github.com/bluesky-social/atproto/pull/2721/commits/7d0deb0a08bb1be5c030e1cb9b1d502d6a6cc3a9#diff-7f86c57c90267a6c6c49cb89dd5233108fd81f59c8de18680fa7b3c1f3893205R141

Reply:
Given `A -> B -> C -> D`, we can even check if `A` blocks `B` based on the data we already hydrate https://github.com/bluesky-social/atproto/pull/2721/files#diff-7f86c57c90267a6c6c49cb89dd5233108fd81f59c8de18680fa7b3c1f3893205R184

Reply:
Actually, just kidding, that's only possible if the grandparent happens to be included in the first page of results. I've restricted the tests to fetching a single `FeedViewPost` now, which therefore matches the frontend, since it uses only a single `FeedViewPost` to construct a `FeedSlice`. We moved to that approach for consistency, instead of _maybe_ having the data we need in the first page of results to do further sorting/munging.

Reply:
>I think actually that in the case of A -> ... -> C -> D, we do hydrate post blocks for A <-> C

Yeah and no on that too, that test was viewing as C so of course A was blocked.

Clarified the block relationships here and fixed up the tests. Triple checking again now to ensure I'm not still confused how this works :P

Reply:
When we hydrate `D` within a feed perhaps we need to ask for three blocking relationships: `(author(D), author(C))`, `(author(A), author(C))`, and `(author(A), author(D))`.  That would ensure that no two of these posts with a 3p block appear in context of each other as a part of the same thread.